construct data as powershell object for proper indentation

jantari opened the pull request:
this also removes the need for the extra step of `ConvertFrom-Json`-ing the here-string because the data is created right away as PowerShell objects.

The results the changed code produces are identical to before

Commit:
construct data as powershell object for proper indentation

also removes the need for the extra step of `ConvertFrom-Json`

bwgjoseph commented:
Thanks for contributing!

That indeed look much better than using `here-string`, learned something new about `PSCustomObject`

Since we are at this topic, would you want to help convert `winget-downloads.ps1` as well? And does this help `vscode-plugin-downloads.ps1` regarding the escaping of quote in `$data_binary`?

Thanks

jantari commented:
Yes I can definitely look into that, do you prefer separate PRs for that?

bwgjoseph commented:
I think we can do on this same PR since it's all related to switching to `PSCustomObject`

Commit:
pwsh object and neater indentation in winget-downloads.ps1

Commit:
fix extract_url regex to work with newer winget versions

also redo the extraction logic so it better utilizes the existing
information in the powershell objects rather than working with
strings

Commit:
use powershell object, pipe json to curl and fix indentation

curl really really wants quotes to be escaped even if it's not
required by the shell itself (e.g. when wrapping in single quotes)
but one way to circumvent that and use JSON directly is to read it
from a file with "@". Using "@-" therefore reads JSON from stdin

jantari commented:
I've pushed some more commits that address the other two files as well.

I noticed when working on `winget-downloads.ps1` that it doesn't actually work, at least on my system with winget v1.4.2161-preview, because the program download URL is returned as `Installer Url:` in the manifest now rather than `Download Url:` so I fixed that in d83746d. It matches either `Download Url:` or `Installer Url:` now so should be backwards-compatible.

bwgjoseph commented:
Great work! I'm using winget `v1.3.2691` and tested it to be working, and the manifest still returns as `Download Url:`. Nice catch there.

If there's nothing else, I think this is ready to be merged in. I prefer a `squash and merge` if you have no objection.

jantari commented:
Of course, no objections!

bwgjoseph commented:
Thanks for your contribution!
